ygot/struct_validation_map.go

```diff
@@ -37,16 +40,27 @@ const (
 	// indentString represents the default indentation string used for
 	// JSON. Three spaces are used based on the legacy use of EmitJSON.
 	indentString string = "   "
+
+	// pathPoolBufSize is the allocation size of the mapPathsPool, this should
+	// not be very large since the size of mapPaths is determined by the number
+	// of "|" separators.
+	pathPoolBufSize = 16
 )
 
+// mapPathsPool helps reduce heap allocations.
+var mapPathsPool = sync.Pool{
```

Comment:
Can you add more details on why this helps avoid heap allocations?

Reply:
This was an experiment in this frequently called function about how much GC load `sync.Pool` can reduce, mostly using reference to the documentation in <https://go.dev/src/sync/pool.go>. In this use case I don't think there was significant GC pressure reduced (much less than the change from switching to `go-json`). If this is overly complex we can revert the use of `sync.Pool` for now. Thanks!